hooks/open-telemetry/pkg/metrics.go

```diff
@@ -25,18 +25,10 @@ type MetricsHook struct {
 	errorCounter   api.Int64Counter
 
 	flagEvalMetadataDimensions []DimensionDescription
+	attributeSetterCallback    *func(openfeature.FlagMetadata) []attribute.KeyValue
```

Comment:
Think we need to support returning errors from the function and logging? I'm not sure, totally up to you if you think that would be valuable enough to implement.

Reply:
I don't think this is needed. Having an error in the callback creates a contract that we need to take care of from SDK end 🤔  IMO this is not needed and error handling must be done by the callback writer.